app/api/tasks/route.ts

```diff
@@ -90,15 +93,26 @@ export async function POST(req: NextRequest) {
       orderBy: { position: 'desc' },
     });
     // Place the new task after the last task
-    const newTaskPosition =
-      lastTask && lastTask?.position ? lastTask.position + 1 : 1;
+    const newTaskPosition = lastTask?.position ? lastTask.position + 1 : 1;
+
     const newTask = await prisma.task.create({
       data: {
         text,
         userId,
         position: newTaskPosition,
       },
     });
+
+    // If tagIds are provided, create TaskTag associations
+    if (tagIds && tagIds.length > 0) {
+      await prisma.taskTag.createMany({
+        data: tagIds.map((tagId: string) => ({
+          taskId: newTask.id,
+          tagId,
+        })),
+      });
+    }
```

Comment:
I created a specific join table because I wanted a way to very quickly gets these associations for graphs and charts in the future - but I might change my mind and get rid of this if it doesn't really serve us, and we can get away with using SQL more concisely.